dist2src/worker/updater.py

```diff
@@ -53,67 +52,65 @@ def check_updates(
-        url = f"{self.cfg.src_git_svc.api_url}projects"
-        params = {
-            "namespace": m["namespace"],
-            "pattern": project,
-            "fork": m["fork"] is not None,
-            "per_page": self.PROJECTS_PER_PAGE,
-            "owner": m["owner"] or None,
-            "short": True,
-        }
-        while url:
-            r = self.cfg.src_git_svc.call_api(url, params=params)
-            # The URL in 'next' already has the required parameters
-            url, params = r["pagination"]["next"], None
-            logger.debug(
-                f"Next page: {url!r}. Total pages: {r['pagination']['pages']!r}."
+        if project:
+            # update only the selected project
+            gitlab_project = self.cfg.src_git_svc.get_project(
+                repo=project, namespace=self.cfg.src_git_namespace
             )
-            for src_git_project in r["projects"]:
-                try:
-                    self._check_project(src_git_project["name"], branch)
-                except (OgrException, RetryError):
-                    logger.exception(
-                        f"Failed checking project {src_git_project['name']!r}"
+            self.check_and_update_project(gitlab_project, branch=branch)
+        else:
+            for n in range(1, self.PROJECTS_PER_PAGE):
+                # Get repositories in source-git
+                projects = src_gitlab_group.projects.list(page=n, per_page=100)
+                if not projects:
+                    break
+                for project_g in projects:
+                    gitlab_project = self.cfg.src_git_svc.get_project(
+                        repo=project_g.name, namespace=self.cfg.src_git_namespace
                     )
-                    continue
+                    self.check_and_update_project(gitlab_project, branch=branch)
         Pushgateway().push_dist2src_finished_checking_updates()
 
-    def _check_project(self, project: str, branch: Optional[str] = None):
-        logger.debug(f"Checking project {project!r}...")
-        dist_git_project = self.cfg.dist_git_svc.get_project(
-            namespace=singular_fork(self.cfg.dist_git_namespace),
-            repo=project,
-            # Specify username in order to disable superfluous whoami API calls.
-            username="packit",
-        )
+    def check_and_update_project(
+        self, project: GitlabProject, branch: Optional[str] = None
+    ):
+        """check selected src repo and queue update tasks for branch which are out of date"""
+        dist_git_project = self._get_dist_git(project)
         if not dist_git_project.exists():
             logger.warning(
                 f"{dist_git_project.full_repo_name!r} does not exist in "
                 f"{self.cfg.dist_git_host!r}"
             )
             Pushgateway().push_found_missing_dist_git_repo()
             return
+        branch_commits = self._get_out_of_date_branches(project, branch)
+        if branch_commits:
```

Comment:
This check seems to be redundant, `_get_out_of_date_branches` will always return a list, even if empty, in which case the for-cycle will do nothing.

Reply:
this is a good point, at some point in my iterations it was returning None, will drop that line, thanks!